tests/src/rules/no-cycle.js

```diff
@@ -30,6 +30,10 @@ ruleTester.run('no-cycle', rule, {
     test({ code: 'var foo = require("@scope/foo")' }),
     test({ code: 'var bar = require("./bar/index")' }),
     test({ code: 'var bar = require("./bar")' }),
+    // correctly passes
+    test({ code: 'import { foo } from "./ignore.js"' }),
+    // incorrectly fails
+    test({ code: 'import { foo } from "./intermediate-ignore.js"' }),
```

Comment:
actually it turns out BOTH of these should be failing - since the tests have `depth-zero` configured as the "current filename" for the `code`, BOTH of them have a cycle, so BOTH of them should be failing!